tiledb/sm/query/writer.cc

```diff
@@ -2451,5 +2843,61 @@ Status Writer::write_tiles(
   return Status::Ok();
 }
 
+// TODO: remove
+Status Writer::zip_coord_tiles(
+    const std::unordered_map<std::string, std::vector<Tile>>& coord_tiles,
+    std::vector<Tile>* coords_tiles) const {
+  if (coord_tiles.empty() || coord_tiles.begin()->second.empty())
+    return Status::Ok();
+
+  auto tile_num = coord_tiles.begin()->second.size();
+  coords_tiles->clear();
+  coords_tiles->resize(tile_num);
+  auto type = array_schema()->domain()->type();
+  unsigned dim_num = array_schema()->dim_num();
+  uint64_t coords_size = array_schema()->coords_size();
+  for (size_t t = 0; t < tile_num; ++t) {
+    auto& new_tile = (*coords_tiles)[t];
+    auto cell_num = coord_tiles.begin()->second[t].cell_num();
+    RETURN_NOT_OK(new_tile.init(
+        constants::format_version,
+        type,
+        cell_num * coords_size,
+        coords_size,
+        dim_num));
+    for (unsigned d = 0; d < dim_num; ++d) {
+      const auto& dim_name = array_schema_->dimension(d)->name();
+      const auto& coord_tile = coord_tiles.find(dim_name)->second[t];
+      new_tile.write(coord_tile.data(), coord_tile.size());
+    }
+  }
+
+  return Status::Ok();
+}
+
+std::string Writer::coords_to_str(uint64_t i) const {
+  std::string ret;
+  auto dim_num = array_schema_->dim_num();
+
+  ret = "(";
```

Comment:
if this will be called frequently, consider a `stringstream` here to avoid a bunch of string copies.

Reply:
No, it isn't, but I will change anyway.